src/Simulation/Core/QArray.cs

```diff
@@ -150,8 +149,7 @@ public void Extend(long newLength)
                 long oldLength = storage.Count;
                 for (int i = 0; i < (newLength - oldLength); i++)
                 {
-                    T obj = CreateDefault();
-                    storage.Add(obj);
+                    storage.Add(Default.OfType<T>());
```

Comment:
While I am sure the compiler should be smart enough to pull this out of the loop, it doesn't hurt to do that manually to make sure it is not evaluated each time.

Reply:
I can do it manually. Since C# is impure and `OfType` could potentially have side effects, I am not so sure that the compiler can perform common subexpression elimination here.